### PR TITLE
chore: improve error messages and logging during shard opening

### PR DESCRIPTION
**Pull request by gwossum**

### CODE REVIEW

File: tsdb/engine/tsm1/file_store.go [hunk 1]
```diff
@@ -561,7 +562,7 @@ func (f *FileStore) Open() error {
 
 		file, err := os.OpenFile(fn, os.O_RDONLY, 0666)
 		if err != nil {
-			return fmt.Errorf("error opening file %s: %v", fn, err)
+			return fmt.Errorf("FileStore.Open: %w", err)
```
Comment (davidby-influx) [inline tsdb/engine/tsm1/file_store.go:565]: Let's keep the file name in the message

File: tsdb/engine/tsm1/file_store.go [hunk 2]
```diff
@@ -573,32 +574,35 @@ func (f *FileStore) Open() error {
 
 			start := time.Now()
 			df, err := NewTSMReader(file, f.readerOptions...)
+
```
Comment (davidby-influx) [inline tsdb/engine/tsm1/file_store.go:577]: unnecessary diff?

File: tsdb/engine/tsm1/file_store.go [hunk 3]
```diff
@@ -573,32 +574,35 @@ func (f *FileStore) Open() error {
...
 			f.logger.Info("Opened file",
 				zap.String("path", file.Name()),
 				zap.Int("id", idx),
 				zap.Duration("duration", time.Since(start)))
 
 			// If we are unable to read a TSM file then log the error.
 			if err != nil {
-				file.Close()
+				if cerr := file.Close(); cerr != nil {
```
Comment (davidby-influx) [inline tsdb/engine/tsm1/file_store.go:585]: I love this.

File: tsdb/engine/tsm1/file_store.go [hunk 4]
```diff
@@ -635,7 +639,7 @@ func (f *FileStore) Open() error {
 			f.lastModified = fi.ModTime().UTC()
 		} else {
 			close(readerC)
-			return err
+			return fmt.Errorf("FileStore.Open: %w", err)
```
Comment (davidby-influx) [inline tsdb/engine/tsm1/file_store.go:642]: Let's add the directory name here.

File: tsdb/index.go [hunk 5]
```diff
@@ -3253,15 +3253,15 @@ func NewIndex(id uint64, database, path string, seriesIDSet *SeriesIDSet, sfile
 	if os.IsNotExist(err) {
 		// nop, use default
 	} else if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("NewIndex: %w", err)
```
Comment (davidby-influx) [inline tsdb/index.go:3256]: Let's add the path in the message

**Review by davidby-influx — CHANGES_REQUESTED** [review]
This will really help us debugging in the future.  Nice work. I'd like to add (or in a few cases) restore the path/file to the error messages as a little more guidance for our debugging.

File: tsdb/engine/tsm1/engine.go [hunk 6]
```diff
@@ -2431,15 +2432,15 @@ func (e *Engine) cleanup() error {
...
+		return fmt.Errorf("Engine.cleanup: %w", err)
 	}
 
 	ext := fmt.Sprintf(".%s", TmpTSMFileExtension)
 	for _, f := range allfiles {
 		// Check to see if there are any `.tmp` directories that were left over from failed shard snapshots
 		if f.IsDir() && strings.HasSuffix(f.Name(), ext) {
 			if err := os.RemoveAll(filepath.Join(e.path, f.Name())); err != nil {
-				return fmt.Errorf("error removing tmp snapshot directory %q: %s", f.Name(), err)
+				return fmt.Errorf("Engine.cleanup: error removing tmp snapshot directory: %w", err)
```
[the remark this answers is not in the document]
Comment (gwossum) [reply tsdb/engine/tsm1/engine.go:2443]: I had that originally, but then I noticed that `PathError` contained the path so I removed the duplicate path. Upon closer inspection, ~these are loafers~ if you get a `SyscallError`, that doesn't include the path.

**Review by davidby-influx — APPROVED** [review]
LGTM
